test/e2e/validation_test.go

```diff
@@ -0,0 +1,197 @@
+		return false, nil
+	})
+	return err
+}
+
+// getInstanceID gets the instanceID of VM for a given cloud provider ID
+// Ex: aws:///us-east-1e/i-078285fdadccb2eaa. We always want the last entry which is the instanceID
+func getInstanceID(providerID string) string {
+	providerTokens := strings.Split(providerID, "/")
+	return providerTokens[len(providerTokens)-1]
+}
+
+// getInstanceIDsOfNode returns the instanceIDs of all the Windows nodes created
+func getInstanceIDsOfNode(kubeclient kubernetes.Interface) ([]string, error) {
+	nodes, err := kubeclient.CoreV1().Nodes().List(metav1.ListOptions{LabelSelector: wmc.WindowsOSLabel})
+	if err != nil {
+		return nil, errors.Wrap(err, "error while querying for Windows nodes")
+	}
+	instanceIDs := make([]string, 0, len(nodes.Items))
+	for _, node := range nodes.Items {
+		if len(node.Spec.ProviderID) > 0 {
+			instanceID := getInstanceID(node.Spec.ProviderID)
+			instanceIDs = append(instanceIDs, instanceID)
+		}
+	}
+	return instanceIDs, nil
+}
+
+// testConfigMapValidation ensures that the required configMap is created and is having appropriate
+// entries
+func testConfigMapValidation(t *testing.T, nodeCount int) {
+	testCtx := framework.NewTestCtx(t)
+	namespace, err := testCtx.GetNamespace()
+	require.NoError(t, err, "error while getting test namespace")
+
+	err = waitForTrackerConfigMap(framework.Global.KubeClient, namespace, nodeCount,
+		retryInterval, time.Minute*1)
+	require.NoError(t, err, "error waiting for tracker configmap")
+
+	// Get the instance id from the cloud provider for the windows Nodes created
+	instanceIDs, err := getInstanceIDsOfNode(framework.Global.KubeClient)
+	require.NoError(t, err, "error while getting provider specific instanceIDs")
+
+	// check if those instances are present in the configmap
+	trackerConfigMap, err := framework.Global.KubeClient.CoreV1().ConfigMaps(namespace).Get(tracker.StoreName, metav1.GetOptions{})
+	require.NoError(t, err, "error while getting the tracker configmap")
+	for _, instanceID := range instanceIDs {
+		require.Contains(t, trackerConfigMap.BinaryData, instanceID)
+	}
+}
+
+// getWindowsVM creates a windowsVM to be used for running commands against
+func getWindowsVM(ipAddress, instanceID string, credentials tracker.Credentials) (types.WindowsVM, error) {
+	winVM := &types.Windows{}
+	windowsCredentials := types.NewCredentials(instanceID, ipAddress, credentials.Password, credentials.Username)
+	winVM.Credentials = windowsCredentials
+	// Set up Winrm client
+	err := winVM.SetupWinRMClient()
+	if err != nil {
+		return nil, errors.Wrap(err, "error instantiating winrm client")
+	}
+	return winVM, nil
+}
+
+// validateConnectivity creates a Windows VM object and ensures that we have connectivity
+// for the Windows VM
+func validateConnectivity(ipAddress, instanceID string, credentials tracker.Credentials) error {
+	windowsVM, err := getWindowsVM(ipAddress, instanceID, credentials)
+	if err != nil {
+		return err
+	}
```

Comment:
Nit: I will be making this call too. Looks like the test struct is going to be required sooner rather than later.

Reply:
Yes, the `getWindowsVM()` call. I can remove the `validateConnectivity()` method if it is no longer useful in my PR though I am fine having explicit test cases.